### PR TITLE
refactor in order to remove spring component

### PR DESCRIPTION
**Pull request by MaelAudren**

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalogclient/service/RemoteObjectService.java [hunk 1]
```diff
@@ -25,35 +25,52 @@
...
+        return response.as(returnClass);
+    }
+
+    /**
+     * Send a GET request on url with a sessionId returning a returnClass object
+     * @param url is the url where the get request will be sent
+     * @param returnClass is the class type of the response
+     * @return a returnClass object created from the server response
+     */
+    public <T> T sendRequest(String url, String sessionId, Class<T> returnClass) {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalogclient/service/RemoteObjectService.java]: MAJOR Remove this unused method parameter "sessionId". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)

File: src/main/java/org/ow2/proactive/catalogclient/service/RemoteObjectService.java [hunk 2]
```diff
@@ -25,35 +25,52 @@
...
 
 
 /**
  * This class enables to send request to remote server
  */
-@Service
 @Log4j2
 public class RemoteObjectService {
 
+    private final static RemoteObjectService REMOTE_OBJECT_SERVICE = new RemoteObjectService();
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalogclient/service/RemoteObjectService.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/main/java/org/ow2/proactive/catalogclient/util/ResponseUtils.java [hunk 3]
```diff
@@ -23,21 +23,18 @@
...
+import org.ow2.proactive.catalogclient.exception.FailedRequestException;
 
+import io.restassured.response.Response;
 
-@ComponentScan(basePackages = { "org.ow2.proactive.catalogclient.*" })
-public class SpringConfig {
 
-    private SpringConfig() {
-    }
+public class ResponseUtils {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalogclient/util/ResponseUtils.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1118)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Review by sophiesongge — APPROVED** [review]

**Review by marcocast — APPROVED** [review]

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**